internal/provider/incident_catalog_entries_data_source.go

```diff
@@ -0,0 +1,187 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/incident-io/terraform-provider-incident/internal/apischema"
+	"github.com/incident-io/terraform-provider-incident/internal/client"
+)
+
+var (
+	_ datasource.DataSource              = &IncidentCatalogEntriesDataSource{}
+	_ datasource.DataSourceWithConfigure = &IncidentCatalogEntriesDataSource{}
+)
+
+func NewIncidentCatalogEntriesDataSource() datasource.DataSource {
+	return &IncidentCatalogEntriesDataSource{}
+}
+
+type IncidentCatalogEntriesDataSource struct {
+	client *client.ClientWithResponses
+}
+
+type IncidentCatalogEntriesDataSourceModel struct {
+	CatalogTypeID  types.String                          `tfsdk:"catalog_type_id"`
+	CatalogEntries []IncidentCatalogEntryDataSourceModel `tfsdk:"catalog_entries"`
```

Comment:
this reuses the incident_catalog_entry_data_source model but happy to be told this is the wrong approach?